### PR TITLE
Add RTGSFIT timing benchmark with GitHub Gist upload

### PR DESCRIPTION
**Pull request by aleksyprok**
## Summary

Adds infrastructure to measure and track RTGSFIT per-section CPU time
across runs, and fixes a bug in the GSFIT coil current data access.

### C library (`src/`)
- Add `rtgsfit_timing_get(double *out_us, int n)` to `rtgsfit.c`/`rtgsfit.h`
  to read accumulated per-section CPU times (µs) into a caller-supplied
  buffer, replacing the stdout-only `rtgsfit_timing_dump()`

### Python (`replay_rtgsfit.py`)
- Register `rtgsfit_timing_get` ctypes signature
- Store per-iteration timing in `output_dict["timing_us"]` (shape `n_iters × 18`)
  alongside a `"timing_labels"` key

### Bug fix (`replay_gsfit.py`)
- Fix coil current key paths from the incorrect
  `["pf", name, "i", "measured_experimental"]` /`"time_experimental"`
  to the correct `["pf", name, "i", "experimental", "value"]` / `"time"`
  (matching the actual `gsfit_rs` DataTree structure)

### New test (`test_timing_rtgsfit.py`)
- Parametrized over all 6 test cases (pulses 13343/13345/13346 × t=0.03/0.10 s)
- Runs 1,000 back-to-back `rtgsfit()` calls from converged state
- Reports mean, median, std, min, max, p95, p99 per section to stdout
- Appends CSV rows to a private GitHub Gist (auto-created/found by
  description); upload silently skipped if `GITHUB_TOKEN` is not set

### Tolerances (`test_integration_rtgsfit_vs_gsfit.py`)
- Loosen `bp_meas` measured tolerances (`rtol_meas` 0.05→0.1,
  `atol_meas` 0.05→0.5) to reflect that RTGSFIT predicted values are
  compared against noisy experimental measurements, not GSFIT predicted

### Docs (`README.md`)
- Add GitHub token setup instructions for the Gist upload feature

## Testing

```bash
pytest -s tests

### CODE REVIEW

File: tests/rtgsfit_vs_gsfit/tests/test_timing_rtgsfit.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+
+def _upload_to_gist(csv_rows: str) -> None:
+    """Append csv_rows to the Gist CSV file, or skip if no token."""
+    import requests  # only imported when actually needed
+
+    token = os.environ.get("GITHUB_TOKEN", "")
+    if not token:
+        print("\n[timing] GITHUB_TOKEN not set — skipping Gist upload.")
+        return
```
[the remark this answers is not in the document]
Comment (aleksyprok) [reply tests/rtgsfit_vs_gsfit/tests/test_timing_rtgsfit.py:86]: Good catch. Fixed by adding `requests` as an explicit dependency in `tests/rtgsfit_vs_gsfit/pyproject.toml` and moving the import to the top of the file, no deferred imports or `try/except` needed.

File: tests/rtgsfit_vs_gsfit/tests/test_timing_rtgsfit.py [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+
+    headers = {"Authorization": f"token {token}", "Accept": "application/vnd.github+json"}
+    gist_id = _find_gist_id(headers)
+
```
[the remark this answers is not in the document]
Comment (aleksyprok) [reply tests/rtgsfit_vs_gsfit/tests/test_timing_rtgsfit.py:90]: Good point. Since we decided RTGSFIT_GIST_ID isn't needed (the Gist is found automatically by description), the fix is to remove it from the docstring rather than re-adding the env var. Docstring updated.

File: src/rtgsfit.c [hunk 3]
```diff
@@ -87,6 +87,14 @@ void rtgsfit_timing_dump(void)
     }
 }
 
+void rtgsfit_timing_get(double *out_us, int n)
+{
+    int count = (n < T_NTIMERS) ? n : T_NTIMERS;
+    for (int i = 0; i < count; i++) {
+        out_us[i] = (double)timing_acc[i] * 1e-3;
+    }
+}
```
[the remark this answers is not in the document]
Comment (aleksyprok) [reply src/rtgsfit.c:96]: Our callers always pass n == T_NTIMERS so these edge cases can't occur in practice. Keeping the function simple.

File: tests/rtgsfit_vs_gsfit/README.md [hunk 4]
```diff
@@ -80,6 +80,23 @@ To run the tests use
...
+**One-time setup:**
+
+1. Create a GitHub Personal Access Token with only the **`gist`** scope at  
+   https://github.com/settings/tokens/new
+
+2. Save it to `~/.bashrc`:
+   ```bash
+   echo 'export GITHUB_TOKEN=ghp_xxxxxxxxxxxx' >> ~/.bashrc
+   source ~/.bashrc
+   ```
```
[the remark this answers is not in the document]
Comment (aleksyprok) [reply tests/rtgsfit_vs_gsfit/README.md:93]: Fair point in general, but this is an internal research tool used on private machines within our organisation. Storing the token in ~/.bashrc is convenient and the risk is acceptable in this context. Keeping as-is.

File: src/rtgsfit.c [hunk 5]
```diff
@@ -87,6 +87,14 @@ void rtgsfit_timing_dump(void)
     }
 }
 
+void rtgsfit_timing_get(double *out_us, int n)
```
Comment (bevincentTE) [inline src/rtgsfit.c:96]: You'd probably want to add a proper header to this function.

File: tests/rtgsfit_vs_gsfit/src/rtgsfit_vs_gsfit/plot/components.py [hunk 6]
```diff
@@ -12,6 +12,9 @@
 import mdsthin
 import numpy as np
 
+def _gsfit_node(cfg: dict, suffix: str) -> str:
+    return f"\\GSFIT::TOP.{cfg['run_name']}.{suffix}"
```
Comment (bevincentTE) [inline tests/rtgsfit_vs_gsfit/src/rtgsfit_vs_gsfit/plot/components.py]: You are ok with the default error message when run_name is not an entry in the dictionary cfg?

File: tests/rtgsfit_vs_gsfit/src/rtgsfit_vs_gsfit/table/dataframes.py [hunk 7]
```diff
@@ -6,6 +6,9 @@
 import numpy as np
 import pandas as pd
 
+def _gsfit_node(cfg: dict, suffix: str) -> str:
```
Comment (bevincentTE) [inline tests/rtgsfit_vs_gsfit/src/rtgsfit_vs_gsfit/table/dataframes.py]: You already defined this function in the file tests/rtgsfit_vs_gsfit/src/rtgsfit_vs_gsfit/plot/components.py. Do you think this redundancy is necessary?

File: tests/rtgsfit_vs_gsfit/tests/test_integration_rtgsfit_vs_gsfit.py [hunk 8]
```diff
@@ -15,6 +15,9 @@
 from rtgsfit_vs_gsfit.plot import examples
 from rtgsfit_vs_gsfit.table import save_to_csv
 
+def _gsfit_node(cfg: dict, suffix: str) -> str:
```
Comment (bevincentTE) [inline tests/rtgsfit_vs_gsfit/tests/test_integration_rtgsfit_vs_gsfit.py]: This function has already been defined before.

**Review by bevincentTE — APPROVED** [review]

File: src/rtgsfit.c (hunk 5 above)
Comment (aleksyprok) [reply src/rtgsfit.c:96]: I added a doc comment to rtgsfit_timing_get.

File: tests/rtgsfit_vs_gsfit/src/rtgsfit_vs_gsfit/plot/components.py (hunk 6 above)
Comment (aleksyprok) [reply tests/rtgsfit_vs_gsfit/src/rtgsfit_vs_gsfit/plot/components.py]: I made gsfit_node() raise a clearer KeyError when run_name is missing.

File: tests/rtgsfit_vs_gsfit/src/rtgsfit_vs_gsfit/table/dataframes.py (hunk 7 above)
Comment (aleksyprok) [reply tests/rtgsfit_vs_gsfit/src/rtgsfit_vs_gsfit/table/dataframes.py]: I moved the helper into a shared module and updated the call sites.

File: tests/rtgsfit_vs_gsfit/tests/test_integration_rtgsfit_vs_gsfit.py (hunk 8 above)
Comment (aleksyprok) [reply tests/rtgsfit_vs_gsfit/tests/test_integration_rtgsfit_vs_gsfit.py]: Same fix as above :)

**Comment by aleksyprok** [comment]
I’ve addressed the run-name normalization and gist truncation issues. I’m leaving the psi tolerance as-is for now because we do still see larger reconstruction errors, so tightening it would make the integration test flaky rather than more useful.

File: tests/rtgsfit_vs_gsfit/src/rtgsfit_vs_gsfit/rtgsfit_compile_setup.py [hunk 9]
```diff
@@ -9,46 +9,10 @@
...
-        except Exception:
-            return False
-    tree_check = MDSplus.Tree("RTGSFIT", cfg["pulse_num_write"])
-    if not node_exists(tree_check, f":{cfg['run_name']}"):
-        deleted = True
-    else:
-        deleted = False
-    assert deleted, f"Node {cfg['run_name']} was not deleted successfully."
-
 def initialise_rtgsfit_node(cfg: dict):
```
[the remark this answers is not in the document]
Comment (aleksyprok) [reply tests/rtgsfit_vs_gsfit/src/rtgsfit_vs_gsfit/rtgsfit_compile_setup.py:16]: I removed the rtgsfit_mds_nodeclear() call from replay_rtgsfit.py and cleaned up the stale README/comment references too.
